src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/runInPodNestedExplicitInherit.groovy

```diff
@@ -2,7 +2,7 @@ podTemplate(label: '$NAME-parent', containers: [
 		containerTemplate(name: 'golang', image: 'golang:1.6.3-alpine', ttyEnabled: true, command: '/bin/cat'),
 	]) {
 
-	podTemplate(inheritFrom: '',  containers: [
+	podTemplate(inheritFrom: '-',  containers: [
```

Comment:
While this would have been a better syntax to mean “do not inherit”, it it too late to do that now—this change is not backward compatible. As mentioned, you need to restore the meaning of `''`, and introduce a different notation for “use default inheritance” which would normally not be present in Groovy but would be used just as a form default for **Pipeline Syntax**.

Reply:
Based on where the inheritFrom is [used](https://github.com/jenkinsci/kubernetes-plugin/blob/25a20a28bbe044cde5eabc32bb0acfa8bda49e0d/src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java#L99), `null` is the way to tell it to "use default inheritance", the change I made for the snippet generator caused the empty string to be converted to null, causing the test to fail. 

I could change the default for `inheritFrom` to be an empty string but I think that would break `runInPodNested` test.

Reply:
> `null` is the way to tell it to "use default inheritance", the change I made for the snippet generator caused the empty string to be converted to null, causing the test to fail.

Correct AFAIK.

> I could change the default for `inheritFrom` to be an empty string

No, that is not safe, as it would disable inheritance by default—even worse than the current state.

The most straightforward solution is to introduce some constant string which is neither blank nor a possible template name and which could reasonably convey the notion of “do the default thing”—say, `<default>`—which would be the default value for the textbox in the form, with `setInheritFrom` translating it to a null field value. This text would never be expected in a real Pipeline script; the snippet generator would simply omit the `inheritFrom` attribute if you did not edit the field, which is what we want. If it does not already, the inline `help-inheritFrom.html` needs to explain the situation, specifically that you may blank out the field to suppress inheritance.

(`SnippetizerTester` is not particularly useful for testing this situation since it makes no reference to the `config.jelly`. `StepConfigTester`, from `workflow-step-api`, actually exercises the config form, though in an artificial way which would also get confused by this trick: `configRoundTrip` would call `getInheritFrom`, see null, and translate that to a blank textfield (the implied `value="${instance.inheritFrom}"` makes no distinct between `null` and `""`), which upon form submission would call the setter on `""` and produce a distinct object. In other words `configRoundTrip` always operates with `instance != null`, whereas the actual **Pipeline Syntax** screen always operates with a null `instance` (only a non-null `descriptor`) since you are not reconfiguring something that is already serialized. Having `getInheritFrom` convert null back to the new magic string would solve that issue, but is not very intuitive, and would require another helper method like `getActualInheritFrom` to be called from Java code, as opposed to the config form. tl;dr: probably best to test behavior of this field interactively.)